pkg/api/v1/deployment.go

```diff
@@ -132,6 +144,44 @@ func (g *DeploymentGroup) StopAllActiveDeployments(ctx echo.Context) error {
 	return g.stopDeployments(deployments, ctx)
 }
 
+func (g *DeploymentGroup) DeleteDeployment(ctx echo.Context) error {
+	cc, ok := ctx.(*auth.HttpAuthContext)
+	if !ok {
+		return echo.NewHTTPError(http.StatusInternalServerError, "Failed to get auth info")
+	}
+
+	deploymentId := ctx.Param("deploymentId")
+
+	// Get deployment
+	deploymentWithRelated, err := g.backendRepo.GetDeploymentByExternalId(ctx.Request().Context(), cc.AuthInfo.Workspace.Id, deploymentId)
+	if err != nil {
+		if err == sql.ErrNoRows {
+			return ctx.JSON(http.StatusNotFound, map[string]interface{}{
+				"error": "deployment not found",
+			})
+		}
+		return ctx.JSON(http.StatusBadRequest, map[string]interface{}{
+			"error": "failed to get deployment",
+		})
+	}
+
+	// Stop deployment first
+	if err := g.stopDeployments([]types.DeploymentWithRelated{*deploymentWithRelated}, ctx); err != nil {
+		return ctx.JSON(http.StatusInternalServerError, map[string]interface{}{
+			"error": "failed to stop deployment",
+		})
+	}
+
+	// Delete deployment
+	if err := g.backendRepo.DeleteDeployment(ctx.Request().Context(), deploymentWithRelated.Deployment); err != nil {
+		return ctx.JSON(http.StatusInternalServerError, map[string]interface{}{
+			"error": "failed to delete deployment",
```

Comment:
Most of these errors start lower cased, but one doesn't. Can we choose a style and keep it consistent?